shared/src/main/scala/coop/rchain/shared/EventLog.scala

```diff
@@ -0,0 +1,26 @@
+package coop.rchain.shared
+
+sealed trait Event
+object Event {
+  object NodeStarted           extends Event
+  object EnteredRunningState   extends Event
+  object ApprovedBlockReceived extends Event
+  object SentUnapprovedBlock   extends Event
+  object BlockApprovalReceived extends Event
+  object SentApprovedBlock     extends Event
```

Comment:
I'd say those should be case classes parameterized with the data allowing for discerning the event instances, and doing debugging.

E.g. any Block-related event should contain its hash
any receive-related eevent should contain the sender

Reply:
I'm also missing a 'Block received' event in here (the above events only cover genesis).

Also, an event should most likely be emitted whenever `.propose` is triggered on a node

Reply:
> E.g. any Block-related event should contain its hash
any receive-related eevent should contain the sender

By event you mean a Scala representation of an event or the thing that gets emitted on a logger?

Reply:
I'd go with the former - the data in the events might be useful for processing otherwise than dumping it into the logs. It also seems prudent to defer string serialization as long as possible.

Reply:
> useful for processing otherwise
not that we're going to store that data now - but having the API be data-bound and not string-bound will make things easier if/when we start storing the data / outputting json

Reply:
@ArturGajowy I augmented the events with the block hash. We can add other events later. For now, this will be used only in the integration tests.